packages/manager/apps/telecom/src/app/telecom/pack/migration/service-delete/pack-migration-service-delete.controller.js

```diff
@@ -11,6 +13,16 @@ export default class TelecomPackMigrationServiceDeleteCtrl {
   $onInit() {
     this.process = null;
     this.process = this.TucPackMigrationProcess.getMigrationProcess();
+    this.process.selectedOffer.subServicesToDelete.forEach((subService) => {
+      set(
+        subService,
+        'services',
+        map(subService.services, (service, index, originalArray) => ({
+          name: service,
+          selected: originalArray.length === subService.numberToDelete,
+        })),
+      );
+    });
```

Comment:
IMO we should avoid using `lodash/set` even if it's done like that in the codebase. Maybe you could use a `map` for that.
And I'm don't get what the `map` is supposed to do 🤔

Reply:
I did this to transform the services array like ['service1', 'service2',...] to array like [{name: 'service1', selected: false}, {name:'service2', selected: false}...] that it could be display into the view.

Reply:
As @cbourgois suggested, this code could be simplified to use at least two map instructions.
```suggestion
    this.process.selectedOffer.subServiceToDelete = this.process.selectedOffer.subServicesToDelete.map((subService) => ({
    ...subService,
    services: services.map(() => // etc...)
    }));
```

Also the logic of the `map`on `subService.services` is quite disturbing. You're setting a `selected` boolean, which will be true if the subService has all its services to be deleted ?

Reply:
this `selected` boolean is for the checkbox displayed into the view of services to delete

Reply:
Not sure, but if you have something like : 

```javascript
subServicesToDelete = [
  {
    services: ["a", "b", "c", "d"],
    numberToDelete: 3,
    // ... props
  },
  {
    services: ["x", "y", "z"],
    numberToDelete: 3,
    // ... props
  }
]
```

The expected result is : 
- `a/b/c/d` service with `selected` set to `false`
- `x/y/z` service with `selected` set to `true`  

Can you confirm ?

Reply:
The only thing I want to know is that we really mark service as `selected` if the parent `numberToDelete` is equal to the number of services in `services`. Strange rule, I don't get it.

Reply:
Yes it's that

Reply:
Ok, let the customer choose which services or select all because all should be deleted. Thanks for the infos.